File: Code/New folder/back/models/location/region.js
const { nanoid } = require('nanoid');
module.exports = (sequelize, DataTypes) => {
    return sequelize.define('Region', {
        regionID: {
            type: DataTypes.STRING,
            primaryKey: true,
            defaultValue: () => `reg_${nanoid(6)}`,
        },
        name: {
            type: DataTypes.STRING,
            unique: true
        },
        nameAr: {
            type: DataTypes.STRING
        },
        nameFr: {
            type: DataTypes.STRING
        }
    });
};

File: Code/Mobile/lib/models/region.dart
class Region {
  final String regionID;
  final String name;
  final String? nameAr;
  final String? nameFr;

  Region({
    required this.regionID,
    required this.name,
    this.nameAr,
    this.nameFr,
  });

  factory Region.fromJson(Map<String, dynamic> json) {
    return Region(
      regionID: json['regionID'] as String,
      name: json['name'] as String,
      nameAr: json['nameAr'] as String?,
      nameFr: json['nameFr'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'regionID': regionID,
      'name': name,
      'nameAr': nameAr,
      'nameFr': nameFr,
    };
  }
}